### PR TITLE
[NGS-1077] Dynamic Blocklist Implementation 

### PR DESCRIPTION
**Pull request by Rajat2809**
## Context

Dynamic Blocklist for each bidder.
It is to implement invalidating any bid in which Bapp and BApp are on the blocklists. Once the bid is invalidated the Bundle and AdDomain will be added to the respective bidder dynamic blocklist BApp and BAdv respectively.

## PRs
TJX: https://github.com/Tapjoy/mediation_bid_service/pull/1525
Go-Prebid: https://github.com/Tapjoy/go-prebid/pull/107
TPE: https://github.com/Tapjoy/tpe_prebid_service/pull/172

 ## How Has This Been Tested?

`./validate.sh` and `make build` yields no error

 ## Jira Link

Jira: https://tapjoy.atlassian.net/browse/NGS-1077

### CODE REVIEW

File: adapters/tjx_aarki/aarki.go [hunk 1]
```diff
@@ -62,6 +65,15 @@ func Builder(_ openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder,
 func (adapter *adapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
 	numRequests := len(request.Imp)
 
+	var srcExt *reqSourceExt
```
Comment (umutdemirel) [inline adapters/tjx_aarki/aarki.go:68]: You don't need pointer here.

File: adapters/tjx_aarki/aarki.go [hunk 2]
```diff
@@ -62,6 +65,15 @@ func Builder(_ openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder,
...
+	if request.Source != nil && request.Source.Ext != nil {
+		if errSrcExt := json.Unmarshal(request.Source.Ext, &srcExt); errSrcExt != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: errSrcExt.Error(),
+			}}
+		}
+	}
```
Comment (umutdemirel) [inline adapters/tjx_aarki/aarki.go:75]: Why do you do this here but not just before you need this below? If you want to do this just one time, then move it to just before the start of a for loop, after `var err error`

And please follow the error convention:

```
if err != nil {
    errs = append(errs, err)
    continue
}
```

File: adapters/tjx_aarki/aarki.go [hunk 3]
```diff
@@ -103,6 +115,19 @@ func (adapter *adapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.Ex
 			continue
 		}
 
+		// This check is for identifyong if request comes from TJX
```
Comment (umutdemirel) [inline adapters/tjx_aarki/aarki.go:118]: fix typo

File: adapters/tjx_rtbhouse/rtbhouse.go [hunk 4]
```diff
@@ -110,14 +123,27 @@ func (adapter *RTBHouseAdapter) MakeRequests(
 			continue
 		}
 
-		var rtbHouseExt openrtb_ext.ExtImpRTBHouse
+		var rtbHouseExt openrtb_ext.ExtImpTJXRTBHouse
```
Comment (umutdemirel) [inline adapters/tjx_rtbhouse/rtbhouse.go:126]: Thanks for the fix

**Review by umutdemirel — CHANGES_REQUESTED** [review]
Put some comments, please apply your changes to all bidders.

File: adapters/tjx_aarki/aarki.go (hunk 1 above)
Comment (Rajat2809) [reply adapters/tjx_aarki/aarki.go:68]: I used it as a pointer for nil check.

**Review by umutdemirel — APPROVED** [review]
LGTM!

**Pull request merged**